src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/HeapHprofBinWriter.java

```diff
@@ -537,7 +537,7 @@ private int calculateClassDumpRecordSize(Klass k) {
             List<Field> instanceFields = new ArrayList<>();
             Iterator<Field> itr = null;
             // loader + signer + protectionDomain + 2 reserved + fieldSize + cpool entris number
-            size += OBJ_ID_SIZE * 5 + INT_SIZE + SHORT_SIZE;
+            size += (int) (OBJ_ID_SIZE * 5 + INT_SIZE + SHORT_SIZE);
```

Comment:
Why are INT_SIZE and SHORT_SIZE longs? It seems that making them int (along with all the other related _SIZE fields) is the proper fix for this and the changes below.

Reply:
These come from ObjectHeap. I do agree it would be better to change these upstream (all the type sizes there are currently `long`), but that would be a much more invasive change. I'm happy to try that, but it looked like a can of worms.

Reply:
Unfortunately that eventually leads to Type.getSize(), which returns a long. I don't think you can change it to return an int, because some types (nonscalar) could need a long to fit their size, so you do need an int cast some point later. Probably the cast should be in the following code since we know all of these fit an in int:

```
  public ObjectHeap(TypeDataBase db) throws WrongTypeException {
    // Get commonly used sizes of basic types
    oopSize     = VM.getVM().getOopSize();
    byteSize    = db.getJByteType().getSize();
    charSize    = db.getJCharType().getSize();
    booleanSize = db.getJBooleanType().getSize();
    intSize     = db.getJIntType().getSize();
    shortSize   = db.getJShortType().getSize();
    longSize    = db.getJLongType().getSize();
    floatSize   = db.getJFloatType().getSize();
    doubleSize  = db.getJDoubleType().getSize();
  }
```

Reply:
I pushed 8115af5 to do as you suggested. That cleaned up most of the casts in HeapHprofBinWriter.